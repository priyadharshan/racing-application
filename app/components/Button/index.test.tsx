import React from "react";
import * as stories from "./index.stories";
import { axe } from "jest-axe";
import userEvent from "@testing-library/user-event";
import { composeStories } from "@storybook/testing-react";
import { render, screen } from "@testing-library/react";

const { Default, Clicked } = composeStories(stories);

describe("Button", () => {
  describe("Default", () => {
    test("should be accessible and match snapshot", async () => {
      const { container } = render(<Default />);
      expect(await axe(container)).toHaveNoViolations();
      expect(container.firstChild).toMatchSnapshot();
    });

    test("should handle click", () => {
      const onClick = jest.fn();
      render(<Default onClick={onClick} />);
      userEvent.click(screen.getByText("Mes références"));
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });

  describe("Clicked", () => {
    test("should handle click", async () => {
      const onClick = jest.fn();
      const { container } = render(<Clicked onClick={onClick} />);
      await Clicked.play({ canvasElement: container });
      expect(onClick).toHaveBeenCalledTimes(1);
    });
  });
});
