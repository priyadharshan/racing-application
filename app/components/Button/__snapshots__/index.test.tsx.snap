// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Default should be accessible and match snapshot 1`] = `
<div
  class="p-4"
>
  <button
    class="btn"
  >
    Mes références
  </button>
</div>
`;
